l10n_it_vat_registries/wizard/print_registro_iva.py

```diff
@@ -0,0 +1,167 @@
+        self.include_rc_moves = self.tax_registry_id.include_rc_moves
+        self.rc_journal_ids = self.tax_registry_id.rc_journal_ids
+
+    @api.onchange("date_range_id")
+    def on_change_date_range_id(self):
+        if self.date_range_id:
+            self.from_date = self.date_range_id.date_start
+            self.to_date = self.date_range_id.date_end
+
+    @api.onchange("from_date")
+    def get_year_footer(self):
+        if self.from_date:
+            self.year_footer = self.from_date.year
+
+    def _get_move_ids_domain(self):
+        return [
+            ("date", ">=", self.from_date),
+            ("date", "<=", self.to_date),
+            ("journal_id", "in", [j.id for j in self.journal_ids]),
+            ("state", "=", "posted"),
+        ]
+
+    def _include_rc_journals(self):
+        if (
+            self.include_rc_moves
+            and self.rc_journal_ids
+            and self.layout_type == "customer"
+        ):
+            return True
+        return False
+
+    def _get_move_ids(self, wizard):
+        MAPPING = {
+            "journal_date_name": "journal_id, date, name",
+            "date_name": "date, name",
+        }
+        order = MAPPING[wizard.entry_order]
+        moves = self.env["account.move"].search(
+            self._get_move_ids_domain(),
+            order=order,
+        )
+        if wizard._include_rc_journals():
+            LAMBDA_MAPPING = {
+                "journal_date_name": lambda m: (m.journal_id, m.date, m.name),
+                "date_name": lambda m: (m.date, m.name),
+            }
+            order = LAMBDA_MAPPING[wizard.entry_order]
+            rc_moves = (
+                self.env["account.move"]
+                .search(
+                    [
+                        ("date", ">=", wizard.from_date),
+                        ("date", "<=", wizard.to_date),
+                        ("journal_id", "in", [j.id for j in wizard.rc_journal_ids]),
+                        ("state", "=", "posted"),
+                    ]
+                )
+                .filtered(lambda m: m.l10n_it_edi_is_self_invoice)
+            )
+            moves |= rc_moves
+            moves = moves.sorted(order)
+        return moves.ids
+
+    def print_registro(self):
+        self.ensure_one()
+        wizard = self
+        if not wizard.journal_ids:
+            raise UserError(
+                _(
+                    "No journals found in the current selection.\n"
+                    "Please load them before to retry!"
+                )
+            )
+        move_ids = self._get_move_ids(wizard)
+
+        datas_form = {}
+        datas_form["from_date"] = wizard.from_date
+        datas_form["to_date"] = wizard.to_date
+        datas_form["journal_ids"] = [j.id for j in wizard.journal_ids]
+        if wizard._include_rc_journals():
+            datas_form["rc_journal_ids"] = [j.id for j in wizard.rc_journal_ids]
```

Comment:
```suggestion
        if wizard._include_rc_journals():
            datas_form["journal_ids"].extend([j.id for j in wizard.rc_journal_ids])
```
Perché non aggiungerli a `journal_ids`? Tanto poi mi sembra vengano messi tutti insieme.

Reply:
Perchè esplicito è meglio di implicito: meglio chiarire che non sono registri normali, ma dedicati a operazioni RC

Reply:
:+1: